src/ngx_http_lua_kong_module.c

```diff
@@ -670,5 +683,197 @@ ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
 
     return ctx->upstream_ssl_verify;
 }
+# endif
+
+char *
+ngx_http_lua_kong_load_var_index(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+
+    ngx_str_t                     *value;
+    ngx_int_t                      index;
+
+    value = cf->args->elts;
+
+    if (value[1].len == 0) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "invalid variable name size \"%V\"",
+                           &value[1]);
+        return NGX_CONF_ERROR;
+    }
+
+    index = ngx_http_get_variable_index(cf, &value[1]);
+
+    if (index == NGX_ERROR) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "undefined variable \"%V\"",
+                           &value[1]);
+        return NGX_CONF_ERROR;
+    }
+
+    return NGX_CONF_OK;
+}
+
+int
+ngx_http_lua_kong_ffi_var_load_indexes(ngx_str_t **names,
+    unsigned int *count, char **err)
+{
+    ngx_uint_t                  i;
+    ngx_http_variable_t        *v;
+    ngx_http_core_main_conf_t  *cmcf;
+
+    cmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle, ngx_http_core_module);
+
+    v = cmcf->variables.elts;
+
+    assert(v != NULL);
```

Comment:
We can use `ngx_http_lua_assert` here, and it will only be enabled for debug builds (Nginx and OpenResty's release build contains no asserts).

See:

https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_common.h#L39-L44
https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_cache.c#L62